diff --git a/iiod-client.c b/iiod-client.c
index fd0986b8..31fcd6e3 100644
--- a/iiod-client.c
+++ b/iiod-client.c
@@ -1946,8 +1946,14 @@ int iiod_client_reg_write(struct iiod_client *client, const struct iio_device *d
 int iiod_client_refresh_format(struct iiod_client *client, const struct iio_device *dev,
 		const struct iio_channel *chn, char *format_str, size_t len)
 {
-	struct iiod_io *io = iiod_responder_get_default_io(client->responder);
-	struct iiod_command cmd = {
+	struct iiod_io *io;
+	struct iiod_command cmd;
+
+	if (!iiod_client_uses_binary_interface(client))
+		return -ENOTSUP;
+
+	io = iiod_responder_get_default_io(client->responder);
+	cmd = (struct iiod_command){
 		.op = IIOD_OP_REFRESH_FORMAT,
 		.dev = (uint8_t)iio_device_get_index(dev),
 		.code = (int32_t)iio_channel_get_index(chn),
